libafl/src/events/launcher.rs

```diff
@@ -234,66 +228,63 @@ where
+                        }
+                        ForkResult::Child => {
+                            // # Safety
+                            // A call to `getpid` is safe.
+                            log::info!("{:?} PostFork", unsafe { libc::getpid() });
+                            self.shmem_provider.post_fork(true)?;
+
+                            std::thread::sleep(Duration::from_millis(index * self.launch_delay));
+
+                            if !debug_output {
+                                if let Some(file) = &self.opened_stdout_file {
+                                    dup2(file.as_raw_fd(), libc::STDOUT_FILENO)?;
+                                    if let Some(stderr) = &self.opened_stderr_file {
+                                        dup2(stderr.as_raw_fd(), libc::STDERR_FILENO)?;
+                                    } else {
+                                        dup2(file.as_raw_fd(), libc::STDERR_FILENO)?;
+                                    }
                                 }
                             }
-                        }
 
-                        // Fuzzer client. keeps retrying the connection to broker till the broker starts
-                        let builder = RestartingMgr::<EMH, MT, S, SP>::builder()
-                            .shmem_provider(self.shmem_provider.clone())
-                            .broker_port(self.broker_port)
-                            .kind(ManagerKind::Client {
-                                cpu_core: Some(*bind_to),
-                            })
-                            .configuration(self.configuration)
-                            .serialize_state(self.serialize_state)
-                            .hooks(hooks);
-                        let builder = builder.time_ref(self.time_ref.clone());
-                        let (state, mgr) = builder.build().launch()?;
-
-                        return (self.run_client.take().unwrap())(state, mgr, *bind_to);
-                    }
-                };
+                            // Fuzzer client. keeps retrying the connection to broker till the broker starts
+                            let builder = RestartingMgr::<EMH, MT, S, SP>::builder()
+                                .shmem_provider(self.shmem_provider.clone())
+                                .broker_port(self.broker_port)
+                                .kind(ManagerKind::Client {
+                                    cpu_core: Some(*bind_to),
+                                })
+                                .configuration(self.configuration)
+                                .serialize_state(self.serialize_state)
+                                .hooks(hooks);
+                            let builder = builder.time_ref(self.time_ref.clone());
+                            let (state, mgr) = builder.build().launch()?;
+
+                            return (self.run_client.take().unwrap())(state, mgr, *bind_to);
```

Comment:
Only issue I see is that this may break clients that assume there's at most one `bind_to` core_id. They may, for example, create a directory based off of the core_id.
We probably want to add the overcommit information somewhere...
Either in the state, in an extra param, or even add the overcommit information into the core id somehow

Reply:
We don't really have access to the client id at this point I think. That's mainly an LLMP thing right now. CoreId is still needed, too, since some fuzzers decide what cores to run ASan on according to the ID, for eaxmple...

Reply:
For now we can just add metadata to the State for example, or see if we can easily add the info to CoreId

Reply:
Ah, I didn't know there was a client id already. I was thinking of creating a new one, that was essentially just a u64, and maybe with some additional data (such as the core id).

Reply:
so yeah store the overcommit in metadata or hack it into the ClientId

Reply:
Also, please add a test for overcommit / use it on a fuzzer / ?